src/ds5/ds5-options.h

```diff
@@ -370,14 +372,89 @@ namespace librealsense
             return "Gain limits ranges from 16 to 248. Default is 0 which means full gain. If the requested gain limit is less than 16, it will be set to 16. If the requested gain limit is greater than 248, it will be set to 248. Setting will not take effect until next streaming session.";
         }
         virtual void enable_recording(std::function<void(const option&)> record_action) override { _record_action = record_action; }
+        float get_cached_limit() { return _cached_limit; };
 
     private:
         std::function<void(const option&)> _record_action = [](const option&) {};
         lazy<option_range> _range;
         hw_monitor& _hwm;
         sensor_base* _sensor;
+        float _cached_limit;
     };
 
+    // Auto-Limits Enable/ Disable
+    class limits_option : public option
+    {
+    public:
+
+        limits_option(rs2_option option, option_range range) : _option(option), _toggle_range(range)
+        {
+            _value = 1;
+        };
+        
+        virtual ~limits_option() = default;
+        virtual void set(float) override = 0;
+        virtual float query() const override { return _value; };
+        virtual option_range get_range() const override { return _toggle_range; };
+        virtual bool is_enabled() const override { return true; }
+        virtual const char* get_description() const override { return "Enable Limits Option"; };
+        virtual void enable_recording(std::function<void(const option&)> record_action) override { _record_action = record_action; }
```

Comment:
most of the override re-implement the base-class functionality.
Needs refactoring